inngest/experimental/dev_server/dev_server.py

```diff
@@ -0,0 +1,117 @@
+# ruff: noqa: S603, S607, T201
+
+import os
+import subprocess
+import threading
+import time
+import typing
+
+import httpx
+
+
+class _Server:
+    @property
+    def origin(self) -> str:
+        return f"http://0.0.0.0:{self.port}"
+
+    def __init__(self) -> None:
+        self._enabled = os.getenv("DEV_SERVER_ENABLED") != "0"
+        self._output_thread: typing.Optional[threading.Thread] = None
+
+        port: int
+        dev_server_port_env_var = os.getenv("DEV_SERVER_PORT")
+        if dev_server_port_env_var:
+            port = int(dev_server_port_env_var)
+        else:
+            port = 8288
+        self.port = port
+
+        self._process: typing.Optional[subprocess.Popen[str]] = None
+        self._ready_event = threading.Event()
+        self._stop_event = threading.Event()
+
+    def start(self) -> None:
+        if self._enabled is False:
+            return
+
+        print("Dev Server: starting")
+
+        if self._process:
+            raise Exception("Dev Server is already running")
+
+        self._process = subprocess.Popen(
+            [
+                "npx",
+                "--yes",
+                "inngest-cli@latest",
+                "dev",
+                "--no-discovery",
+                "--no-poll",
+                "--port",
+                f"{self.port}",
+            ],
+            bufsize=1,
+            stderr=subprocess.STDOUT,
+            stdout=subprocess.PIPE,
+            text=True,
+            universal_newlines=True,
+        )
```

Comment:
Might be good to move away from `npx` for the Python SDK when this comes out of experimental - `npm` could be an extra tool folks have to install.

Not sure what we've move to, though, outside of managing a `/tmp/` version where we install the binary directly.